### PR TITLE
User GET API By UserId

### PR DESCRIPTION
**Pull request by EXR-97**

### CODE REVIEW

File: src/main/java/com/elixr/poc/constants/ApplicationConstants.java [hunk 1]
```diff
@@ -17,6 +17,4 @@ public class ApplicationConstants {
     public static final String AMOUNT_MISSING_IN_PURCHASE = "Amount is mandatory";
     public static final String PRODUCT_NAME_IS_MISSING_IN_PURCHASE = "Product name are mandatory";
     public static final String DATE_MISSING_IN_PURCHASE = "Date are mandatory";
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/constants/ApplicationConstants.java]: grammatical mistake "are"

File: src/main/java/com/elixr/poc/constants/ApplicationConstants.java [hunk 2]
```diff
@@ -17,6 +17,4 @@ public class ApplicationConstants {
     public static final String AMOUNT_MISSING_IN_PURCHASE = "Amount is mandatory";
     public static final String PRODUCT_NAME_IS_MISSING_IN_PURCHASE = "Product name are mandatory";
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/constants/ApplicationConstants.java]: grammatical mistake "are"

File: src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    private final UserService userService;
+
+    public UserRetrievalController(UserService userService) {
+        this.userService = userService;
+    }
+
+    /**
+     * Getting the user by userId.
+     * Calling a method getUserByUserId of service class.
+     * Printing the user and HttpStatus.
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java]: probably this statement is wrong. We are not printing anywhere

File: src/main/java/com/elixr/poc/service/UserService.java [hunk 4]
```diff
@@ -40,6 +41,7 @@ public boolean deleteUserDetails(UUID userId) throws NoRecordFoundException {
 
     /**
      * Creating a valid user
+     *
```
Comment (mh-saicharanvelury) [inline src/main/java/com/elixr/poc/service/UserService.java:45]: remove these unwanted empty lines

File: src/main/java/com/elixr/poc/constants/ApplicationConstants.java (hunk 1 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/constants/ApplicationConstants.java]: Updated.

File: src/main/java/com/elixr/poc/constants/ApplicationConstants.java (hunk 2 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/constants/ApplicationConstants.java]: Updated.

File: src/main/java/com/elixr/poc/service/UserService.java (hunk 4 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/service/UserService.java:45]: Removed.

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java [hunk 5]
```diff
@@ -24,23 +21,20 @@ public UserDeletionController(UserService userService) {
...
-            deleteResponse.setErrorMessage(ApplicationConstants.SUCCESSFULLY_DELETED);
+            commonErrorResponse = CommonErrorResponse.builder().success(success).errorMessage(ApplicationConstants.SUCCESSFULLY_DELETED).build();
             httpStatus = HttpStatus.OK;
-        } catch (NoRecordFoundException e) {
-            deleteResponse.setSuccess(false);
-            deleteResponse.setErrorMessage(e.getMessage());
-            httpStatus = HttpStatus.BAD_REQUEST;
+        } catch (Throwable throwable) {
+            throw new GlobalException(ApplicationConstants.ID_MISMATCH);
         }
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java]: why are you catching throwable?

File: src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java [hunk 6]
```diff
@@ -20,8 +28,26 @@ public UserRetrievalController(UserService userService) {
...
+    public ResponseEntity retrieveUser(@PathVariable("userId") UUID userId) throws GlobalException {
+        HttpStatus httpStatus;
+        try {
+            User user = userService.getUserByUserId(userId);
+            UserResponse userResponse = UserResponse.builder().success(true).id(user.getId()).userName(user.getUserName())
+                    .firstName(user.getFirstName()).lastName(user.getLastName()).build();
+            httpStatus = HttpStatus.OK;
+            return new ResponseEntity<>(userResponse, httpStatus);
+        } catch (Throwable throwable) {
+            throw new GlobalException(ApplicationConstants.ID_MISMATCH);
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java]: Why are you catching the throwable?

**Review by mh-spunnappala — CHANGES_REQUESTED** [review]

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java (hunk 5 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java]: Replaced with GlobalException.

File: src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java (hunk 6 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java]: Replaced with GlobalException.

File: src/main/java/com/elixr/poc/constant/ApplicationConstants.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
+package com.elixr.poc.constant;
+
+/* this constant file holds all the error messages for all the end points */
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ApplicationConstants {
```
Comment (mh-spillai) [inline src/main/java/com/elixr/poc/constant/ApplicationConstants.java]: You can move this to message.properties and read it from there. Validation messages should be in the message,.properties.

File: src/main/java/com/elixr/poc/constant/ApplicationConstants.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+    public static final String FIRST_NAME_MISSING_IN_USER = "First name is mandatory";
+    public static final String USER_NAME_MISSING_IN_USER = "User name is mandatory";
+    public static final String LAST_NAME_MISSING_IN_USER = "Last name is mandatory";
```
Comment (mh-scarmel) [inline src/main/java/com/elixr/poc/constant/ApplicationConstants.java]: add alphabetic order

File: src/main/java/com/elixr/poc/constants/ApplicationConstants.java [hunk 9]
```diff
@@ -1,21 +1,21 @@
...
+    public static final String ID_MISMATCH = "The ID is Not Matching";
     public static final String LAST_NAME_MISSING_IN_USER = "Last name is mandatory";
+    public static final String PRODUCT_NAME_IS_MISSING_IN_PURCHASE = "Product name is mandatory";
     public static final String SUCCESSFULLY_DELETED = "Successfully Deleted";
-    public static final String ID_MISMATCH = "The ID is Not Matching";
-    public static final String AMOUNT_MISSING_IN_PURCHASE = "Amount is mandatory";
-    public static final String PRODUCT_NAME_IS_MISSING_IN_PURCHASE = "Product name are mandatory";
-    public static final String DATE_MISSING_IN_PURCHASE = "Date are mandatory";
-     public static final String DELETED = "Deleted";
+    public static final String USER_NAME_MISSING = "User name is mandatory";
```
Comment (mh-scarmel) [inline src/main/java/com/elixr/poc/constants/ApplicationConstants.java]: All constant variable should be aphabetic order

File: src/main/java/com/elixr/poc/constants/ApplicationConstants.java [hunk 10]
```diff
@@ -1,21 +1,21 @@
 package com.elixr.poc.constants;
 
 /**
- * This constant file holds all the error messages for all the end point
+ * This constant file holds all the error messages for all the end points
  */
+
```
Comment (mh-spillai) [inline src/main/java/com/elixr/poc/constants/ApplicationConstants.java]: Why two separate constant files?

**Review by mh-spillai — CHANGES_REQUESTED** [review]

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java [hunk 11]
```diff
@@ -24,23 +24,16 @@ public UserDeletionController(UserService userService) {
...
-    Calling deleteUserDetails method with the parameter userId to delete the user by userId.
-    And handling the Exception if the userId is not matching.
+     * Calling deleteUserDetails method with the parameter userId to delete the user by userId.
+     * And handling the Exception if the userId is not matching.
      */
     @DeleteMapping("/user/{userId}")
-    public ResponseEntity<?> deleteUser(@PathVariable("userId") UUID userId) {
-        DeleteResponse deleteResponse = new DeleteResponse();
+    public ResponseEntity deleteUser(@PathVariable("userId") @Valid UUID userId) {
+        CommonErrorResponse commonErrorResponse;
```
Comment (mh-scarmel) [inline src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java]: Better to create CommonResponse and  success case not need to add errormessage jues message

File: src/main/java/com/elixr/poc/data/User.java [hunk 12]
```diff
@@ -8,7 +8,7 @@
 import java.util.UUID;
 
 /**
```
Comment (mh-saicharanvelury) [inline src/main/java/com/elixr/poc/data/User.java:10]: Except the same comment line coming one line down compared to previous state, there is no change in this file. You can roll back this back to previous state unless someone has commented on it

**Review by mh-saicharanvelury — CHANGES_REQUESTED** [review]
Added one comment

File: src/main/java/com/elixr/poc/service/UserService.java [hunk 13]
```diff
@@ -21,31 +20,33 @@ public class UserService {
...
-    public boolean deleteUserDetails(UUID userId) throws NoRecordFoundException {
+    public boolean deleteUserDetails(UUID userId) throws GlobalException {
         boolean success = false;
         boolean userRecordExists = userRepository.existsById(userId);
         if (userRecordExists) {
             userRepository.deleteById(userId);
             success = true;
         } else {
-            throw new NoRecordFoundException(ApplicationConstants.ID_MISMATCH);
+            throw new GlobalException(ApplicationConstants.ID_MISMATCH);
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/service/UserService.java]: Error message not matching with the specific use case. Here the scenario is the id does not exist, not the id mismatch. Also, the GlobalException class is a custom exception class and it is better to rename it to be specific to your application

File: src/main/java/com/elixr/poc/service/UserService.java [hunk 14]
```diff
@@ -55,19 +56,23 @@ public ErrorResponse createValidUser(User user) {
...
+     * Finding User by userId and returning the user.
+     * @param userId
      * @return
+     * @throws GlobalException
      */
-    public GetAllResponse getAllUsers() {
-        return GetAllResponse.builder().success(true).users(userRepository.findAll()).build();
+    public User getUserByUserId(UUID userId) throws GlobalException {
+        Optional<User> user = userRepository.findById(userId);
+        return user.orElseThrow(() -> new GlobalException(ApplicationConstants.ID_MISMATCH));
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/service/UserService.java]: The error message should match the use case.

**Review by mh-spunnappala — CHANGES_REQUESTED** [review]

File: src/main/java/com/elixr/poc/constant/ApplicationConstants.java (hunk 8 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/constant/ApplicationConstants.java]: Updated

File: src/main/java/com/elixr/poc/constants/ApplicationConstants.java (hunk 9 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/constants/ApplicationConstants.java]: Updated

File: src/main/java/com/elixr/poc/constant/ApplicationConstants.java (hunk 7 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/constant/ApplicationConstants.java]: Updated as per your suggestion

File: src/main/java/com/elixr/poc/constants/ApplicationConstants.java (hunk 10 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/constants/ApplicationConstants.java]: Removed one file

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java (hunk 11 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java]: Updated the 1st part of the comment, but as per requirement we need to show message as "errorMessage"

File: src/main/java/com/elixr/poc/data/User.java (hunk 12 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/data/User.java:10]: Yes the commented lines are the changes made by others, to remove the conflicts I just add that line.

File: src/main/java/com/elixr/poc/service/UserService.java (hunk 13 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/service/UserService.java]: Updated as suggested

File: src/main/java/com/elixr/poc/service/UserService.java (hunk 14 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/service/UserService.java]: Updated

File: src/main/java/com/elixr/poc/common/MessagesKeyEnum.java [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+public enum MessagesKeyEnum {
+    ENTITY_DELETED_SUCCESSFULLY("entity.deleted.successfully"),
+    ENTITY_ID_DOES_NOT_EXISTS("entity.id.does.not.exists"),
+    ENTITY_MANDATORY_FIELD_MISSING("entity.mandatory.field.missing"),
+    ENTITY_USER_EXISTS("entity.user.exists");
+
+    MessagesKeyEnum(String key) {
+        this.key=key;
+    }
+    public String key;
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/common/MessagesKeyEnum.java]: Lower the visibility of this field

File: src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java [hunk 16]
```diff
@@ -0,0 +1,61 @@
+package com.elixr.poc.common.exception;
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java:1]: Why there are two global exception handler class?

File: src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+public class GlobalExceptionHandler {
+
+    /**
+     * Handling the Exception and sending error message
+     *
+     * @param methodArgumentNotValidException
+     * @return
+     */
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    private ResponseEntity handleMethodArgumentNotValidException(MethodArgumentNotValidException methodArgumentNotValidException) {
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java]: Provide the parameterized type for this generic

File: src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    /**
+     * URL exceptions are handled.
+     *
+     * @param idNotFoundException
+     * @return
+     */
+    @ExceptionHandler(IdNotFoundException.class)
+    public ResponseEntity handleGlobalException(IdNotFoundException idNotFoundException) {
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java]: Provide the parameterized type for this generic

File: src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    /**
+     * Generic exceptions are handled.
+     *
+     * @param exception
+     * @return
+     */
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity handleGenericException(Exception exception) {
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java]: Provide the parameterized type for this generic

File: src/main/resources/messages.properties [hunk 20]
```diff
@@ -0,0 +1,4 @@
+entity.deleted.successfully = Delete Successfully.
+entity.id.does.not.exists = ID Does Not Exists.
+entity.mandatory.field.missing = is missing.
+entity.user.exists = User already exists.
```
Comment (mh-spunnappala) [inline src/main/resources/messages.properties]: Missing new line

**Review by mh-spunnappala — CHANGES_REQUESTED** [review]

File: src/main/java/com/elixr/poc/common/MessagesKeyEnum.java (hunk 15 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/common/MessagesKeyEnum.java]: Updated.

File: src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java (hunk 16 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java:1]: By mistake I committed old file too, I removed the old file

File: src/main/resources/messages.properties (hunk 20 above)
Comment (EXR-97) [reply src/main/resources/messages.properties]: Updated

File: src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java (hunk 17 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java]: Updated.

File: src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java (hunk 18 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java]: Updated.

File: src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    /**
+     * URL exceptions are handled.
+     *
+     * @param idNotFoundException
+     * @return
+     */
+    @ExceptionHandler(IdNotFoundException.class)
+    public ResponseEntity<CommonResponse> handleGlobalException(IdNotFoundException idNotFoundException) {
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java]: Rename the method to match the use case

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java [hunk 22]
```diff
@@ -1,19 +1,19 @@
...
-import com.elixr.poc.rest.response.DeleteResponse;
+import com.elixr.poc.common.MessagesKeyEnum;
+import com.elixr.poc.common.util.MessagesUtil;
+import com.elixr.poc.rest.response.CommonResponse;
 import com.elixr.poc.service.UserService;
+import jakarta.validation.Valid;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
 import org.springframework.web.bind.annotation.*;
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java]: wildcard import not recommended

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java [hunk 23]
```diff
@@ -1,19 +1,19 @@
...
 
 import java.util.UUID;
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java]: Remove unused imports

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java [hunk 24]
```diff
@@ -24,23 +24,17 @@ public UserDeletionController(UserService userService) {
...
-    And handling the Exception if the userId is not matching.
+     * Calling deleteUserDetails method with the parameter userId to delete the user by userId.
+     * And handling the Exception if the userId is not matching.
      */
     @DeleteMapping("/user/{userId}")
-    public ResponseEntity<?> deleteUser(@PathVariable("userId") UUID userId) {
-        DeleteResponse deleteResponse = new DeleteResponse();
+    public ResponseEntity deleteUser(@PathVariable("userId") @Valid String userId) {
+        CommonResponse commonErrorResponse;
         HttpStatus httpStatus;
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java]: Remove this local declaration of HttpStatus, instead directly pass the value

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java [hunk 25]
```diff
@@ -24,23 +24,17 @@ public UserDeletionController(UserService userService) {
...
-    Calling deleteUserDetails method with the parameter userId to delete the user by userId.
-    And handling the Exception if the userId is not matching.
+     * Calling deleteUserDetails method with the parameter userId to delete the user by userId.
+     * And handling the Exception if the userId is not matching.
      */
     @DeleteMapping("/user/{userId}")
-    public ResponseEntity<?> deleteUser(@PathVariable("userId") UUID userId) {
-        DeleteResponse deleteResponse = new DeleteResponse();
+    public ResponseEntity deleteUser(@PathVariable("userId") @Valid String userId) {
+        CommonResponse commonErrorResponse;
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java]: Remove this declaration here and assign the value directly

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java [hunk 26]
```diff
@@ -24,23 +24,17 @@ public UserDeletionController(UserService userService) {
     }
 
     /**
-    Calling deleteUserDetails method with the parameter userId to delete the user by userId.
-    And handling the Exception if the userId is not matching.
+     * Calling deleteUserDetails method with the parameter userId to delete the user by userId.
+     * And handling the Exception if the userId is not matching.
      */
     @DeleteMapping("/user/{userId}")
-    public ResponseEntity<?> deleteUser(@PathVariable("userId") UUID userId) {
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java:31]: Format the method

File: src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java [hunk 27]
```diff
@@ -1,16 +1,26 @@
...
+import jakarta.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.UUID;
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java]: Remove unused imports

File: src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java [hunk 28]
```diff
@@ -20,8 +30,21 @@ public UserRetrievalController(UserService userService) {
...
+     * Calling a method getUserByUserId of service class.
+     * Returning the response.
+     * And handling the Exception if the userId is not matching.
+     *
+     * @param userId
+     * @return
+     * @throws IdNotFoundException
+     */
+    @GetMapping("/user/{userId}")
+    public ResponseEntity retrieveUser(@PathVariable("userId") @Valid String userId) {
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java]: Format the method

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java [hunk 29]
```diff
@@ -24,23 +24,17 @@ public UserDeletionController(UserService userService) {
...
-        DeleteResponse deleteResponse = new DeleteResponse();
+    public ResponseEntity deleteUser(@PathVariable("userId") @Valid String userId) {
+        CommonResponse commonErrorResponse;
         HttpStatus httpStatus;
-        try {
             boolean success = userService.deleteUserDetails(userId);
-            deleteResponse.setSuccess(success);
-            deleteResponse.setErrorMessage(ApplicationConstants.SUCCESSFULLY_DELETED);
+            commonErrorResponse = CommonResponse.builder().success(success)
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java]: Why the variable name is like commonErrorResponse even if the process is successful? and why success message is set to errorMessage?

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java (hunk 26 above)
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java:31]: Provide the parameterized tye for this generic

File: src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java (hunk 28 above)
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java]: Provide the parameterized tye for this generic

File: src/main/java/com/elixr/poc/service/UserService.java [hunk 30]
```diff
@@ -21,53 +22,77 @@ public class UserService {
...
-            userRepository.deleteById(userId);
+            userRepository.deleteById(uuid);
             success = true;
         } else {
-            throw new NoRecordFoundException(ApplicationConstants.ID_MISMATCH);
+            throw new IdNotFoundException(MessagesUtil.getMessage(MessagesKeyEnum.ENTITY_ID_DOES_NOT_EXISTS.getKey()));
         }
-        return true;
+        return success;
     }
```
Comment (mh-spunnappala) [inline src/main/java/com/elixr/poc/service/UserService.java]: Refactor this method, consider the following 
1. early return the exception message if the user not found
2. Can avoid using the success flag

**Review by mh-spunnappala — CHANGES_REQUESTED** [review]

File: src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java (hunk 21 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java]: Updated.

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java (hunk 22 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java]: Updated.

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java (hunk 25 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java]: Updated

File: src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java (hunk 28 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java]: Updated

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java (hunk 29 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java]: Solved

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java (hunk 26 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java:31]: Updated

File: src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java (hunk 28 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java]: Updated

File: src/main/java/com/elixr/poc/service/UserService.java (hunk 30 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/service/UserService.java]: Updated.

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java (hunk 26 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java:31]: Updated

File: src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java (hunk 23 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/rest/controller/UserDeletionController.java]: Updated.

**Review by mh-spunnappala — APPROVED** [review]

**Review by mh-spillai — APPROVED** [review]

**Pull request merged**